### PR TITLE
Support GitHub mirrors in CMake build configurations

### LINKED ISSUE #622: Enhancement: Support GitHub mirrors in CMake build configurations
### Search before asking

- [X] I had searched in the [issues](https://github.com/apache/incubator-kvrocks/issues) and found no similar issues.

### Motivation

GitHub has some access issues in some countries (connection resets or slow speeds), 
so we can support custom URL of dependencies in CMake configurations, such as `ghproxy.com`, 
which will speed up build procedure, e.g. `./build.sh build -DGITHUB_PROXY=https://ghproxy.com`.

**NOTICE: we may need to add SHA verification of fetched files to ensure that users are not vulnerable to MITM attacks by public proxy providers if we decide to add this feature.**

### Solution

_No response_

### Are you willing to submit a PR?

- [X] I'm willing to submit a PR!

### PR DESCRIPTION
**Pull request by PragmaTwice**
Close #622.

### CODE REVIEW

**Review by git-hulk — DISMISSED** [review]

**Review by tisonkun — DISMISSED** [review]
LGTM. @PragmaTwice I think we can add a short section in README.md for consumers of this feature.

**Comment by PragmaTwice** [comment]
> LGTM. @PragmaTwice I think we can add a short section in README.md for consumers of this feature.

Thanks for your suggestion, done. @tisonkun

**Review by git-hulk — DISMISSED** [review]

File: README.md [hunk 1]
```diff
@@ -102,8 +102,10 @@ $ ./unittest
 
 ### Supported platforms
 
-* CentOS 6 or 7
-* Ubuntu
+* Linux distributions
+  * CentOS 6 or 7
+  * Ubuntu
+  * and most other distros
```
Comment (PragmaTwice) [inline README.md:109]: Some other changes, not sure whether should move to a seperate PR.

File: README.md [hunk 2]
```diff
@@ -68,7 +68,7 @@ It is as simple as:
 ```shell
 $ git clone https://github.com/apache/incubator-kvrocks.git
 $ cd incubator-kvrocks
-$ ./build.sh build # `./build.sh -h` to check more options
+$ ./build.sh build # `./build.sh -h` to check more options, and `--ghproxy` is recommanded for chinese users
```
Comment (tisonkun) [inline README.md]:
```suggestion
$ ./build.sh build # `./build.sh -h` to check more options;
                   # especially, `./build.sh build --ghproxy` will fetch dependencies via ghproxy.com.
```

File: README.md (hunk 1 above)
Comment (tisonkun) [reply README.md:109]: I'm OK with this change.

**Review by tisonkun — APPROVED** [review]

**Comment by tisonkun** [comment]
We may need to find some queuing strategy to reduce update branch overhead :P

**Comment by PragmaTwice** [comment]
> We may need to find some queuing strategy to reduce update branch overhead :P

I think we can also use ccache and actions/cache to reduce build time in CI by incremental build. 
Although this will make the workflow more complicated.

**Pull request merged**